internal/payloads/launch_template_payload.go

```diff
@@ -14,28 +15,21 @@ type LaunchTemplateResponse struct {
 }
 
 type LaunchTemplateListResponse struct {
-	Data []*LaunchTemplateResponse `json:"data" yaml:"data"`
-}
-
-func (s *LaunchTemplateResponse) Bind(_ *http.Request) error {
-	return nil
-}
-
-func (s *LaunchTemplateResponse) Render(_ http.ResponseWriter, _ *http.Request) error {
-	return nil
+	Data     []*LaunchTemplateResponse `json:"data" yaml:"data"`
+	Metadata page.Metadata             `json:"metadata" yaml:"metadata"`
 }
 
 func (s *LaunchTemplateListResponse) Render(_ http.ResponseWriter, _ *http.Request) error {
 	return nil
 }
 
-func NewListLaunchTemplateResponse(sl []*clients.LaunchTemplate) render.Renderer {
+func NewListLaunchTemplateResponse(sl []*clients.LaunchTemplate, meta *page.Metadata) render.Renderer {
 	list := make([]*LaunchTemplateResponse, len(sl))
-	for i, instanceType := range sl {
+	for i, tmpl := range sl {
 		list[i] = &LaunchTemplateResponse{
-			ID:   instanceType.ID,
-			Name: instanceType.Name,
+			ID:   tmpl.ID,
+			Name: tmpl.Name,
 		}
 	}
-	return &LaunchTemplateListResponse{Data: list}
+	return &LaunchTemplateListResponse{Data: list, Metadata: *meta}
```

Comment:
I dont think it is needed, we are always returning a Metadata struct if you take a look here: `NewTokenMetadata` and `NewOffsetMetadata`
there is not a case we return nil from those functions.

Reply:
I understand. This was just an idea for better maintaining in the future.

The point is, if someone uses this function in the future, they do not have to know that it does not handle nil. Best functions are the ones of which implementation you do not have to know. :laughing: 
If someone wanted to call the function in the future without `NewListLaunchTemplateResponse` Metadata parameter, they would have to create an empty one anyway and pass a pointer to such an empty Metadata structure. If this had been done in the codebase more times, it would create a duplication.
In a worse case, they would think that the function accepts a nil pointer. They would find out pretty quickly because of the panic, but yeah, that was just why I thought it might be handy.